src/tests/JIT/Regression/JitBlue/Runtime_105621/Runtime_105621.cs

```diff
@@ -35,4 +39,114 @@ public static void TestShiftToZero()
+        {
+            var vr3 = Vector64.Create<byte>(128);
+            var vr4 = AdvSimd.ShiftRightLogical(vr3, getByteImmOOB());
+            Assert.Equal(vr4, Vector64<byte>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightLogicalScalarByZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(1);
+            var vr4 = AdvSimd.ShiftRightLogicalScalar(vr3, 0);
+            Assert.Equal(vr3, vr4);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightLogicalScalarToZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(128);
+            var vr4 = AdvSimd.ShiftRightLogicalScalar(vr3, 65);
+            Assert.Equal(vr4, Vector64<long>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightLogicalScalarToZeroNonConst()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(128);
+            var vr4 = AdvSimd.ShiftRightLogicalScalar(vr3, getLongImmOOB());
+            Assert.Equal(vr4, Vector64<long>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticByZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector128.Create<short>(1);
+            var vr4 = AdvSimd.ShiftRightArithmetic(vr3, 0);
+            Assert.Equal(vr3, vr4);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticToZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector128.Create<short>(128);
+            var vr4 = AdvSimd.ShiftRightArithmetic(vr3, 17);
+            Assert.Equal(vr4, Vector128<short>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticToZeroNonConst()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector128.Create<short>(128);
+            var vr4 = AdvSimd.ShiftRightArithmetic(vr3, getShortImmOOB());
+            Assert.Equal(vr4, Vector128<short>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticScalarByZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(1);
+            var vr4 = AdvSimd.ShiftRightArithmeticScalar(vr3, 0);
+            Assert.Equal(vr3, vr4);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticScalarToZero()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(128);
+            var vr4 = AdvSimd.ShiftRightArithmeticScalar(vr3, 65);
+            Assert.Equal(vr4, Vector64<long>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestShiftRightArithmeticScalarToZeroNonConst()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr3 = Vector64.Create<long>(128);
+            var vr4 = AdvSimd.ShiftRightArithmeticScalar(vr3, getLongImmOOB());
```

Comment:
Could you also add equivalent tests for `ShiftLeftLogical` and `ShiftLeftLogicalScalar`?

It should help ensure we don't have `+1` more issues to handle from some missed edge case and will help ensure correct behavior in any future refactorings.

Reply:
Sure thing, looks like `ShiftLeft*` suffers from the same issues we ran into initially with `ShiftRight*`. In Debug, we don't use `ShiftLogical` as a fallback intrinsic, and we end up throwing an exception for out-of-range immediates. In Release, we constant-fold `GT_LSH` operands, but we don't allow over-shifting on ARM64, so the constant-folded behavior doesn't match what `ShiftLogical` would do. I'll add the fallback implementation in this PR to fix this in one go.